lnwallet/channel.go

```diff
@@ -4648,7 +4648,12 @@ func (lc *LightningChannel) SignNextCommitment() (*NewCommitState, error) {
 			newCommitView.txn,
 		)
 		if err != nil {
-			close(cancelChan)
+			select {
```

Comment:
I took another look at how exactly the `cancelChan` is used here and in `tapd`. And I think the original idea was that it would only ever be closed in `SignNextCommitment()`. And only directly before a `return`, so it was guaranteed to only happen once.
And in any other place than `SignNextCommitment`, we would only ever select on the channel to abort (since that would be in a goroutine).

Therefore, I don't think we need to change anything here, really. We just need to follow the same pattern in `tapd` and never `close()` the channel ourselves but instead only select on it whenever we block somewhere (e.g. when writing to `sigJob.Resp`.

I think the description of the `Cancel` channel is what got me to implement this incorrectly in `tapd`:
https://github.com/lightningnetwork/lnd/blob/4960eadd4f179e62b2e8fd4df9f9a84ba69ed3b0/lnwallet/aux_signer.go#L53

```
	// Cancel is a channel that should be closed if the caller wishes to
	// abandon all pending sign jobs part of a single batch.
```

This should read something like this:

```
	// Cancel is a channel that is closed by the caller if they wish to
	// abandon all pending sign jobs part of a single batch. This should
	// never be closed by the validator.
```

And maybe we should make it `<-chan struct{}` to disallow it being closed in the first place by the verify job?

Reply:
Those changes on the `tapd` since were merged with https://github.com/lightninglabs/taproot-assets/pull/1118 .